src/main/java/com/lishid/openinv/ConfigUpdater.java

```diff
@@ -0,0 +1,135 @@
+package com.lishid.openinv;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+import org.bukkit.Material;
+import org.bukkit.configuration.ConfigurationSection;
+import org.bukkit.configuration.file.FileConfiguration;
+
+import com.lishid.openinv.utils.UUIDUtil;
+
+public class ConfigUpdater {
+    private final OpenInv plugin;
+
+    private static final int CONFIG_VERSION = 2;
+
+    public ConfigUpdater(OpenInv plugin) {
+        this.plugin = plugin;
+    }
+
+    private int getConfigVersion() {
+        return plugin.getConfig().getInt("config-version", 1);
+    }
+
+    private boolean isConfigOutdated() {
+        return getConfigVersion() < CONFIG_VERSION;
+    }
+
+    public void checkForUpdates() {
+        if (isConfigOutdated()) {
+            plugin.getLogger().info("[Config] Update found! Performing update...");
+            performUpdate();
+        } else {
+            plugin.getLogger().info("[Config] Update not found. Config is already up-to-date.");
+        }
+    }
+
+    private void performUpdate() {
+        // Update according to the right version
+        switch (getConfigVersion()) {
+            case 1:
+                updateConfig1To2();
+                break;
```

Comment:
I'm imagining this will eventually become something like this, is that what you're thinking as well?

case 1:
 upgradeConfig1to2();
 //No break, falls through directly
case 2:
 upgradeConfig2to3();
etc...